### ISSUE TITLE
After update to 2.7.0 some GOG games doesn't launch and all my library folder was erased

### ISSUE THREAD

**Issue by cann357** [Heroic-Games-Launcher/HeroicGamesLauncher#2633]
### Describe the bug

Heroic after update to 2.7.0 some GOG games doesn't launch and my entire library folder was erased, i was testing some methods to make run postal classic and uncut from GOG and it didn't ever so messing with the options i touched move library folder but then hit to "cancel" cause reallized than that wasn't the option i was looking for so, after cancel and the window was closed, then it started to say that any game was available on my library with yellow prompted game icons and when i closed app then open it again it was saying the same and then tried to run a random game when it said that looks like the installation folder was deleted and when i went to the file browser to check it, it was really deleted, i'm really sad, those were 800gb of games not some much for some people in fact i've not half of my library installed but the really sad thing is that my internet ain't the best (1mbps days, 4mbs nights) and i've to download everything again the funny thing is that i hit cancel button never the move or continue button as well, so thats it, have a nice day but please try if u can to fix that issue to anyone ealse suffer thru like that, even more people like me with this crapy internet. 

### Add logs

```text
(00:11:41) INFO:    [Connection]:       Connectivity: check-online
(00:11:41) INFO:    [Connection]:       Pinging external endpoints
(00:11:41) INFO:    [Legendary]:        Running command: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/legendary --version
(00:11:41) INFO:    [Legendary]:        Legendary location: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/legendary
(00:11:41) INFO:    [Gog]:              GOGDL location: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/gogdl
(00:11:41) INFO:    [Gog]:              GOGDL location: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/gogdl
(00:11:42) WARNING: [Backend]:          Protocol already registered.
(00:11:42) INFO:    [Gog]:              Running command: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/cann/.config/heroic/gog_store/auth.json --version
(00:11:42) INFO:    [Connection]:       Connectivity: online
(00:11:42) INFO:    [Gog]:              Getting data about the user
(00:11:42) INFO:    [Gog]:              Running command: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/cann/.config/heroic/gog_store/auth.json auth
(00:11:43) INFO:    [Backend]:          

Heroic Version: 2.7.0 Eustass Kid
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.7.1

Electron Version: 23.1.4
Chrome Version: 110.0.5481.192
NodeJS Version: 18.12.1

OS: Fedora Linux KERNEL: 6.2.9-200.fc37.x86_64 ARCH: x64
CPU: AMD FX(tm)-8120 Eight-Core Processor @3.1 GOVERNOR: ondemand
RAM: Total: 7.71 GiB Available: 3.16 GiB
GRAPHICS: GPU0: GP106 [GeForce GTX 1060 3GB]  VRAM: 32MB 
PROTOCOL: x11

(00:11:43) INFO:    [Backend]:          Loading Screen Ready
(00:11:43) INFO:    [Frontend]:         Refreshing Library
(00:11:43) INFO:    [Legendary]:        Refreshing library...
(00:11:43) INFO:    [Legendary]:        Refreshing Epic Games...
(00:11:43) DEBUG:   [Legendary]:        Update is available for 56a31432931740cdb0112d237d7d65aa: 0.1.9.65252 != 1.0.4.76104
(00:11:43) DEBUG:   [Legendary]:        Update is available for 70f92598b33d42e290e411f07397aeee: 1.1.9a-10763-343 != 1.2.1-10780-357
(00:11:43) DEBUG:   [Legendary]:        Update is available for 7179f095efcc4f92a950a6d9dbd9c602: 3.3.8 != 3.4.24
(00:11:43) DEBUG:   [Legendary]:        Update is available for AzaleaAlpha: PROSPECT-Releases-R2.6.x-221216.1605_Shipping-CL_1401266 != R3.1.2
(00:11:43) DEBUG:   [Legendary]:        Update is available for Chicken: 2.3.2.4 != 2.17.2
(00:11:43) DEBUG:   [Legendary]:        Update is available for Finch: 473305-2022_12_01_04h53m04s != 484021-2023_03_01_04h58m30s
(00:11:43) DEBUG:   [Legendary]:        Update is available for Jackal: 1.12.2.402612 != 1.13.5.545604
(00:11:43) DEBUG:   [Legendary]:        Update is available for Pewee: 1623 != 1698
(00:11:43) DEBUG:   [Legendary]:        Update is available for Pigeon: 1.1 != 1.2
(00:11:43) DEBUG:   [Legendary]:        Update is available for Sugar: BC2_Update40+407707 != BC2_Update41+415510
(00:11:43) INFO:    [Legendary]:        Found 10 games to update
(00:11:43) INFO:    [Legendary]:        Game list updated, got 79 games & DLCs
(00:11:43) INFO:    [Gog]:              Running command: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/cann/.config/heroic/gog_store/auth.json auth
(00:11:43) INFO:    [Backend]:          Frontend Ready
(00:11:43) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(00:11:43) INFO:    [Backend]:          Checking for new Heroic Updates
(00:11:43) INFO:    [Backend]:          Checking for current version changelog
(00:11:44) INFO:    [Gog]:              Getting GOG library
(00:11:44) INFO:    [Winetricks]:       Downloading Winetricks
(00:11:44) INFO:    [Legendary]:        Running command: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(00:11:44) INFO:    [Gog]:              Saved user data to config
(00:11:48) INFO:    [Gog]:              Number of library pages: 1
(00:11:48) INFO:    [Gog]:              Running command: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/cann/.config/heroic/gog_store/auth.json auth
(00:11:48) INFO:    [Backend]:          Starting the Download Queue
(00:11:52) WARNING: [Backend]:          No clientId in goggame-1825786806.info file. Cannot resolve save path
(00:11:52) INFO:    [Gog]:              Saved games data
(00:11:57) INFO:    [Gog]:              Found 1 game(s) to update
(00:13:20) INFO:    [Frontend]:         Refreshing Library
(00:13:20) INFO:    [Legendary]:        Refreshing library...
(00:13:20) INFO:    [Legendary]:        Refreshing Epic Games...
(00:13:20) DEBUG:   [Legendary]:        Update is available for 56a31432931740cdb0112d237d7d65aa: 0.1.9.65252 != 1.0.4.76104
(00:13:20) DEBUG:   [Legendary]:        Update is available for 70f92598b33d42e290e411f07397aeee: 1.1.9a-10763-343 != 1.2.1-10780-357
(00:13:20) DEBUG:   [Legendary]:        Update is available for 7179f095efcc4f92a950a6d9dbd9c602: 3.3.8 != 3.4.24
(00:13:20) DEBUG:   [Legendary]:        Update is available for AzaleaAlpha: PROSPECT-Releases-R2.6.x-221216.1605_Shipping-CL_1401266 != R3.1.2
(00:13:20) DEBUG:   [Legendary]:        Update is available for Chicken: 2.3.2.4 != 2.17.2
(00:13:20) DEBUG:   [Legendary]:        Update is available for Finch: 473305-2022_12_01_04h53m04s != 484021-2023_03_01_04h58m30s
(00:13:20) DEBUG:   [Legendary]:        Update is available for Jackal: 1.12.2.402612 != 1.13.5.545604
(00:13:20) DEBUG:   [Legendary]:        Update is available for Pewee: 1623 != 1698
(00:13:20) DEBUG:   [Legendary]:        Update is available for Pigeon: 1.1 != 1.2
(00:13:20) DEBUG:   [Legendary]:        Update is available for Sugar: BC2_Update40+407707 != BC2_Update41+415510
(00:13:20) INFO:    [Legendary]:        Found 10 games to update
(00:13:20) INFO:    [Legendary]:        Game list updated, got 79 games & DLCs
(00:13:20) INFO:    [Gog]:              Running command: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/cann/.config/heroic/gog_store/auth.json auth
(00:13:20) INFO:    [Gog]:              Getting GOG library
(00:13:20) INFO:    [Legendary]:        Running command: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/legendary list --third-party
(00:13:21) INFO:    [Gog]:              Found 1 game(s) to update
(00:13:21) INFO:    [Gog]:              Number of library pages: 1
(00:13:21) INFO:    [Gog]:              Running command: /usr/lib64/heroic-games-launcher-bin/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/cann/.config/heroic/gog_store/auth.json auth
(00:13:23) WARNING: [Backend]:          No clientId in goggame-1825786806.info file. Cannot resolve save path
(00:13:23) INFO:    [Gog]:              Saved games data
(00:13:42) INFO:    [Backend]:          Checking for current version changelog
```

### Steps to reproduce

1. Go to Library
2. Click on any installed game and go to move option
3. click to move then cancel
5. See the error

### Expected behavior

Click to move then cancel and really cancel with do any change

### Heroic Version

Latest Stable

### System Information

- OS Fedora 37

**Comment by SteveMunday** [comment]
+1 This happened to me too.  Attaching my log as well if it's helpful.

```
(09:18:01) INFO:    [Connection]:       Connectivity: check-online
(09:18:01) INFO:    [Connection]:       Pinging external endpoints
(09:18:02) INFO:    [Legendary]:        Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(09:18:02) INFO:    [Legendary]:        Legendary location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(09:18:02) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(09:18:02) INFO:    [Gog]:              GOGDL location: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(09:18:02) WARNING: [Backend]:          Failed to register protocol with OS.
(09:18:02) INFO:    [Gog]:              Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json --version
(09:18:02) INFO:    [Connection]:       Connectivity: online
(09:18:02) INFO:    [Backend]:          User Not Found, removing it from Store
(09:18:02) INFO:    [Gog]:              Getting data about the user
(09:18:02) INFO:    [Gog]:              Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json auth
(09:18:02) INFO:    [Backend]:          Loading Screen Ready
(09:18:02) INFO:    [Frontend]:         Refreshing Library
(09:18:02) INFO:    [Legendary]:        Refreshing library...
(09:18:02) WARNING: [Legendary]:        Failed to parse games from /home/deck/.var/app/com.heroicgameslauncher.hgl/config/legendary/installed.json with: Error: ENOENT: no such file or directory, open '/home/deck/.var/app/com.heroicgameslauncher.hgl/config/legendary/installed.json'
    at Object.openSync (node:fs:600:3)
    at Object.func [as openSync] (node:electron/js2c/asar_bundle:2:1822)
    at Object.readFileSync (node:fs:468:35)
    at t.readFileSync (node:electron/js2c/asar_bundle:2:9165)
    at LegendaryLibrary.listUpdateableGames (/app/bin/heroic/resources/app.asar/build/electron/main.a23a7115.js:6:10658)
    at /app/bin/heroic/resources/app.asar/build/electron/main.a23a7115.js:191:3977
    at node:electron/js2c/browser_init:2:97909
    at EventEmitter.<anonymous> (node:electron/js2c/browser_init:2:81422)
    at EventEmitter.emit (node:events:513:28)
(09:18:02) WARNING: [Legendary]:        Failed to parse games from /home/deck/.var/app/com.heroicgameslauncher.hgl/config/legendary/assets.json with: Error: ENOENT: no such file or directory, open '/home/deck/.var/app/com.heroicgameslauncher.hgl/config/legendary/assets.json'
    at Object.openSync (node:fs:600:3)
    at Object.func [as openSync] (node:electron/js2c/asar_bundle:2:1822)
    at Object.readFileSync (node:fs:468:35)
    at t.readFileSync (node:electron/js2c/asar_bundle:2:9165)
    at LegendaryLibrary.listUpdateableGames (/app/bin/heroic/resources/app.asar/build/electron/main.a23a7115.js:6:10978)
    at /app/bin/heroic/resources/app.asar/build/electron/main.a23a7115.js:191:3977
    at node:electron/js2c/browser_init:2:97909
    at EventEmitter.<anonymous> (node:electron/js2c/browser_init:2:81422)
    at EventEmitter.emit (node:events:513:28)
(09:18:02) INFO:    [Legendary]:        Found 0 games to update
(09:18:02) INFO:    [Gog]:              Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json auth
(09:18:02) INFO:    [Backend]:          Frontend Ready
(09:18:03) INFO:    [Backend]:          Checking for current version changelog
(09:18:03) INFO:    [Backend]:          

Heroic Version: 2.7.0 Eustass Kid
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.7.1

Electron Version: 23.1.4
Chrome Version: 110.0.5481.192
NodeJS Version: 18.12.1

OS: Freedesktop SDK KERNEL: 5.13.0-valve36-1-neptune ARCH: x64
CPU: AMD Custom APU 0405 @3.28 GOVERNOR: schedutil
RAM: Total: 14.47 GiB Available: 10.57 GiB
GRAPHICS: GPU0: VanGogh [AMD Custom GPU 0405] VRAM: 512MB 
PROTOCOL: x11

(09:18:03) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(09:18:04) INFO:    [Winetricks]:       Downloading Winetricks
(09:18:05) INFO:    [Gog]:              Found 1 game(s) to update
(09:18:05) INFO:    [Frontend]:         No cache found, getting data from legendary...
(09:18:05) INFO:    [Legendary]:        Refreshing library...
(09:18:07) INFO:    [Backend]:          Starting the Download Queue
(09:18:36) INFO:    [Frontend]:         Refreshing Library
(09:18:36) INFO:    [Legendary]:        Refreshing library...
(09:18:36) INFO:    [Gog]:              Running command: /app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/deck/.var/app/com.heroicgameslauncher.hgl/config/heroic/gog_store/auth.json auth
(09:18:37) INFO:    [Frontend]:         No cache found, getting data from legendary...
(09:18:37) INFO:    [Legendary]:        Refreshing library...
```

**Comment by eddebc** [comment]
+1. Attaching logs while trying to launch CP2077 (running it without Heroic works flawlessly).   
After not launching Heroic in a long time, I was asked to re-auth. This also repopulated my game list as it was incorrectly empty for a while.  
  
While playing around in the game settings I noticed that the game saves location wasn't set and save sync can't be enabled (although it works on an older Heroic on my Steamdeck). Not sure if it's related to saves but a recent large PR changed the way game install info is pulled so my _wild_ guess is _maybe_.

```
(22:00:15) INFO:    [Connection]:       Connectivity: check-online
(22:00:15) INFO:    [Connection]:       Pinging external endpoints
(22:00:15) INFO:    [Legendary]:        Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary --version
(22:00:15) INFO:    [Legendary]:        Legendary location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(22:00:15) INFO:    [Gog]:              GOGDL location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(22:00:15) INFO:    [Gog]:              GOGDL location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
APPIMAGE env is not defined, current application is not an AppImage
(22:00:15) WARNING: [Backend]:          Protocol already registered.
(22:00:15) INFO:    [Gog]:              Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/edd/.config/heroic/gog_store/auth.json --version
(22:00:15) INFO:    [Connection]:       Connectivity: online
(22:00:15) INFO:    [Backend]:          User Not Found, removing it from Store
(22:00:15) INFO:    [Gog]:              Getting data about the user
(22:00:15) INFO:    [Gog]:              Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/edd/.config/heroic/gog_store/auth.json auth
/bin/sh: line 1: hostname: command not found
(22:00:16) INFO:    [Backend]:          

Heroic Version: 2.7.0 Eustass Kid
Legendary Version:  0.20.32 Dark Energy (hotfix #6)
GOGdl Version: 0.7.1

Electron Version: 23.1.4
Chrome Version: 110.0.5481.192
NodeJS Version: 18.12.1

OS: Arch KERNEL: 6.2.10-zen1-1-zen ARCH: x64
CPU: AMD Ryzen 7 5800X 8-Core Processor @3.8 GOVERNOR: schedutil
RAM: Total: 31.26 GiB Available: 15.65 GiB
GRAPHICS: GPU0: Navi 21 [Radeon RX 6800/6800 XT / 6900 XT] VRAM: 1MB 
PROTOCOL: x11

(22:00:16) INFO:    [Backend]:          Loading Screen Ready
(22:00:16) INFO:    [Frontend]:         Refreshing Library
(22:00:16) INFO:    [Legendary]:        Refreshing library...
(22:00:16) WARNING: [Legendary]:        Failed to parse games from /home/edd/.config/legendary/installed.json with: Error: ENOENT: no such file or directory, open '/home/edd/.config/legendary/installed.json'
    at Object.openSync (node:fs:600:3)
    at Object.func [as openSync] (node:electron/js2c/asar_bundle:2:1822)
    at Object.readFileSync (node:fs:468:35)
    at t.readFileSync (node:electron/js2c/asar_bundle:2:9165)
    at LegendaryLibrary.listUpdateableGames (/opt/Heroic/resources/app.asar/build/electron/main.a23a7115.js:6:10658)
    at /opt/Heroic/resources/app.asar/build/electron/main.a23a7115.js:191:3977
    at node:electron/js2c/browser_init:2:97909
    at EventEmitter.<anonymous> (node:electron/js2c/browser_init:2:81422)
    at EventEmitter.emit (node:events:513:28) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/home/edd/.config/legendary/installed.json'
}
(22:00:16) WARNING: [Legendary]:        Failed to parse games from /home/edd/.config/legendary/assets.json with: Error: ENOENT: no such file or directory, open '/home/edd/.config/legendary/assets.json'
    at Object.openSync (node:fs:600:3)
    at Object.func [as openSync] (node:electron/js2c/asar_bundle:2:1822)
    at Object.readFileSync (node:fs:468:35)
    at t.readFileSync (node:electron/js2c/asar_bundle:2:9165)
    at LegendaryLibrary.listUpdateableGames (/opt/Heroic/resources/app.asar/build/electron/main.a23a7115.js:6:10978)
    at /opt/Heroic/resources/app.asar/build/electron/main.a23a7115.js:191:3977
    at node:electron/js2c/browser_init:2:97909
    at EventEmitter.<anonymous> (node:electron/js2c/browser_init:2:81422)
    at EventEmitter.emit (node:events:513:28) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/home/edd/.config/legendary/assets.json'
}
(22:00:16) INFO:    [Legendary]:        Found 0 games to update
(22:00:16) INFO:    [Gog]:              Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/edd/.config/heroic/gog_store/auth.json auth
(22:00:16) INFO:    [Backend]:          Frontend Ready
(22:00:16) INFO:    [Backend]:          Checking for new Heroic Updates
(22:00:16) INFO:    [Backend]:          Checking for current version changelog
(22:00:16) INFO:    [Gog]:              Getting GOG library
(22:00:16) INFO:    [Backend]:          AreWeAntiCheatYet data downloaded
(22:00:16) INFO:    [Gog]:              Saved user data to config
(22:00:16) INFO:    [Gog]:              Number of library pages: 3
(22:00:16) INFO:    [Gog]:              Getting data for page 2
(22:00:17) INFO:    [Gog]:              Found 0 game(s) to update
(22:00:17) INFO:    [Frontend]:         No cache found, getting data from legendary...
(22:00:17) INFO:    [Legendary]:        Refreshing library...
(22:00:17) INFO:    [Gog]:              Getting data for page 3
(22:00:18) INFO:    [Gog]:              Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/edd/.config/heroic/gog_store/auth.json auth
(22:00:18) INFO:    [Winetricks]:       Downloading Winetricks
(22:00:18) INFO:    [Gog]:              Getting data for page 2
(22:00:19) INFO:    [Gog]:              Saved games data
(22:00:20) ERROR:   [ExtraGameInfo]:    Was not able to get GamesDB data for 1980301910
(22:00:20) ERROR:   [ExtraGameInfo]:    Was not able to get GamesDB data for 1486144755
(22:00:21) INFO:    [Backend]:          Starting the Download Queue
(22:00:22) INFO:    [ExtraGameInfo]:    Using cached ExtraGameInfo data for Cyberpunk 2077
(22:00:22) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:22) INFO:    [ExtraGameInfo]:    Using cached ExtraGameInfo data for Cyberpunk 2077
(22:00:22) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:22) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:22) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:25) INFO:    [Backend]:          Launching Cyberpunk 2077 (1423049311)
(22:00:25) INFO:    [Backend]:          Preventing display from sleep
(22:00:25) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-Proton7-43
(22:00:25) INFO:    [Backend]:          Preventing machine to sleep
(22:00:25) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:25) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:26) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-Proton7-43
(22:00:26) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-Proton7-43
(22:00:26) INFO:    [Backend]:          Checking if wine version exists: Wine - Wine-GE-Proton7-43
(22:00:26) DEBUG:   [Backend]:          Running Wine command: wineboot --init
(22:00:26) INFO:    [DXVKInstaller]:    installing dxvk on... /home/edd/Games/Heroic/Prefixes/Cyberpunk 2077
(22:00:26) INFO:    [DXVKInstaller]:    dxvk already installed!
(22:00:26) INFO:    [DXVKInstaller]:    installing vkd3d on... /home/edd/Games/Heroic/Prefixes/Cyberpunk 2077
(22:00:26) INFO:    [DXVKInstaller]:    vkd3d already installed!
(22:00:26) INFO:    [Gog]:              Launching Cyberpunk 2077: LD_PRELOAD= WINEPREFIX="/home/edd/Games/Heroic/Prefixes/Cyberpunk 2077" WINEDLLOVERRIDES=winemenubuilder= ORIG_LD_LIBRARY_PATH= LD_LIBRARY_PATH=/home/edd/.config/heroic/tools/wine/Wine-GE-Proton7-43/lib64:/home/edd/.config/heroic/tools/wine/Wine-GE-Proton7-43/lib GST_PLUGIN_SYSTEM_PATH_1_0=/home/edd/.config/heroic/tools/wine/Wine-GE-Proton7-43/lib64/gstreamer-1.0:/home/edd/.config/heroic/tools/wine/Wine-GE-Proton7-43/lib/gstreamer-1.0 WINEDLLPATH=/home/edd/.config/heroic/tools/wine/Wine-GE-Proton7-43/lib64/wine:/home/edd/.config/heroic/tools/wine/Wine-GE-Proton7-43/lib/wine /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/edd/.config/heroic/gog_store/auth.json launch /home/edd/Games/Heroic 1423049311 --wine /home/edd/.config/heroic/tools/wine/Wine-GE-Proton7-43/bin/wine --platform windows
(22:00:26) INFO:    [Backend]:          Stopping Power Saver Blocker
(22:00:26) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:26) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:26) INFO:    [Backend]:          Stopping Display Power Saver Blocker
(22:00:26) INFO:    [Frontend]:         Refreshing Library
(22:00:26) INFO:    [Gog]:              Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/edd/.config/heroic/gog_store/auth.json auth
(22:00:26) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:26) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:27) INFO:    [Gog]:              Getting GOG library
(22:00:27) INFO:    [Gog]:              Number of library pages: 3
(22:00:27) INFO:    [Gog]:              Getting data for page 2
(22:00:28) INFO:    [Gog]:              Getting data for page 3
(22:00:29) INFO:    [Gog]:              Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/edd/.config/hero
ic/gog_store/auth.json auth
(22:00:29) INFO:    [Gog]:              Getting data for page 2
(22:00:30) INFO:    [Gog]:              Saved games data
(22:00:30) INFO:    [Frontend]:         No cache found, getting data from legendary...
(22:00:30) INFO:    [Legendary]:        Refreshing library...
(22:00:30) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:30) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:00:30) ERROR:   [ExtraGameInfo]:    Was not able to get GamesDB data for 1980301910
(22:00:30) ERROR:   [ExtraGameInfo]:    Was not able to get GamesDB data for 1486144755
(22:01:10) INFO:    [Gog]:              Got install info from cache for 1423049311 on windows platform
(22:01:10) WARNING: [Backend]:          No clientId in goggame-1423049311.info file. Cannot resolve save path
(22:01:10) ERROR:   [Backend]:          gog_save_location/install_path undefined. gog_save_location =  undefined install_path =  /home/edd/Games/Heroic
(22:01:10) INFO:    [Backend]:          1423049311: Setting gogSaves to [{"name":"","location":""}]
```
When I set a save location through the GUI (manually editing the conf file does nothing) and try to save
```
(22:06:40) INFO:    [Backend]:          1423049311: Setting gogSaves to [{"name":"","location":"/home/edd/Games/Heroic"}]
(22:06:46) INFO:    [Gog]:              Running command: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/edd/.config/heroic/gog_store/auth.json auth
(22:06:47) INFO:    [Gog]:              Syncing saves for Cyberpunk 2077
(22:06:47) INFO:    [Gog]:              Syncing saves for Cyberpunk 2077: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl --auth-config-path /home/edd/.config/heroic/gog_store/auth.json save-sync /home/edd/Games/Heroic 1423049311 --os windows --ts 0 --name --skip-upload
```
```
gogdl save-sync: error: argument --name: expected one argument
```
The field `clientId` exists in the `.info` file.
```
edd@pureblue ~ % grep clientId "./Games/Heroic/Cyberpunk 2077/goggame-1423049311.info"
    "clientId": "53652320436400298",
```
The function that [tests](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/48d2787af4079e4365624a38baf1238287d13a62/src/backend/storeManagers/gog/library.ts#L63) this condition, is probably concatenating the [wrong path (executable rather than installation)](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/48d2787af4079e4365624a38baf1238287d13a62/src/backend/storeManagers/gog/library.ts#L601). The "get installed game info" was [changed](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/commit/48d2787af4079e4365624a38baf1238287d13a62#diff-75c581806411cc3a9836fe2ab8a48326e994f2e0693653b70920ca340ec82555L188) recently in a large refactor.  
  
Again, _wild_ guess.

**Comment by arielj** [comment]
I understand this issue should be fixed in Heroic 2.9.1? there was an issue with the install path for GOG games that caused the wrong path to be considered when uninstalling a game, deleting the wrong folder.

Also the `argument` error got fixed in 2.9.1 too

**Comment by imLinguin** [comment]
Indeed this is fixed, also if anyone still encounters the issue, it might be required to modify install path in `.config/heroic/gog_store/installed.json` to point to correct directory
